### ISSUE TITLE
关于Distribute分布式训练的问题

### ISSUE THREAD

**Issue by peach-xiao** [Lsan2401/RMSIN#13]
作者您好！我没有用单机多卡的分布式进行训练过代码，我自己也在网上看了很多，感觉有点总说纷纭的感觉，没有找到解决方案。或者您哪里有比较标准的分布式训练教程可以分享一下吗？我现在有一些问题，或许在您看来比较简单，但是我确实没能找到确切的答案，所以恳请您帮我回答一下：

1.为了让结果可以浮现，使用 torch.distributed进行单机多卡的分布式训练，不需要设置随机种子吗？代码中没有设置，神经网络的参数都是随机初始化的，那么这样的话每次运行的结果不是都不一样吗？

示例如下：
os.environ['PYTHONHASHSEED'] = str(seed)
torch.manual_seed(seed)
torch.cuda.manual_seed_all(seed)
np.random.seed(seed)
random.seed(seed)
torch.backends.cudnn.deterministic = True
torch.backends.cudnn.benchmark = False

2.在计算loss进行反向传播的时候，由于torch.distributed只自动执行了梯度的全局同步，并没有对loss进行全局同步，所以只能看到单卡的loss，看到全局平均loss需要通过all_reduce实现。反向传播的时候不需要先计算全局平均loss，在执行loss.backward()吗？，或者说这个全局平均loss仅仅是用于查看训练集和验证集的loss，和模型的梯度优下降无关？

示例如下：
def reduce_tensor(tensor: torch.Tensor):
    rt = tensor.clone()
    torch.distributed.all_reduce(rt, op=distributed.ReduceOp.SUM)
    rt /= torch.distributed.get_world_size()#总进程数
    return rt

# calculate loss
loss = criterion(predict, labels)
reduced_loss = reduce_tensor(loss.data)
train_epoch_loss += reduced_loss.item()
